Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp

```diff
@@ -318,4 +320,134 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_string_tag_getter)
     return js_string(vm, static_cast<TypedArrayBase&>(this_object).element_name());
 }
 
+// 23.2.3.23 %TypedArray%.prototype.set ( source [ , offset ] ), https://tc39.es/ecma262/#sec-%typedarray%.prototype.set
+JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::set)
+{
+    auto* typed_array = typed_array_from(vm, global_object);
+    if (!typed_array)
+        return {};
+
+    auto source = vm.argument(0);
+
+    auto target_offset = vm.argument(1).to_integer_or_infinity(global_object);
+    if (vm.exception())
+        return {};
+
+    if (target_offset < 0) {
+        vm.throw_exception<JS::RangeError>(global_object);
+        return {};
+    }
+
+    if (source.is_object() && is<TypedArrayBase>(source.as_object())) {
+        auto& source_typed_array = static_cast<TypedArrayBase&>(source.as_object());
+        // 23.2.3.23.1 SetTypedArrayFromTypedArray ( target, targetOffset, source ), https://tc39.es/ecma262/#sec-settypedarrayfromtypedarray
+        auto target_buffer = typed_array->viewed_array_buffer();
+        if (target_buffer->is_detached()) {
+            vm.throw_exception<JS::TypeError>(global_object);
+            return {};
+        }
+        auto target_length = typed_array->array_length();
+        auto target_byte_offset = typed_array->byte_offset();
+
+        auto source_buffer = source_typed_array.viewed_array_buffer();
+        if (source_buffer->is_detached()) {
+            vm.throw_exception<JS::TypeError>(global_object);
+            return {};
+        }
+        auto source_length = source_typed_array.array_length();
+        auto source_byte_offset = source_typed_array.byte_offset();
+
+        if (isinf(target_offset)) {
+            vm.throw_exception<JS::RangeError>(global_object);
+            return {};
+        }
+
+        if (source_length + target_offset > target_length) {
```

Comment:
this looks suspiciously like it could overflow (slap a FIXME on it if you agree but don't want to deal with it right now)

Reply:
I kinda stopped thinking about it and just started translating the spec at that point, I'll add a `Checked` here.